src/Microsoft.Tye.Hosting/Model/Application.cs

```diff
@@ -97,10 +97,32 @@ void SetBinding(ServiceDescription targetService, ServiceBinding b)
                     envName = $"{serviceName.ToUpper()}_{b.Name.ToUpper()}";
                 }
 
+                // Use the container name as the host name if there's a single replica (current limitation)
+                var host = b.Host ?? (service.Description.RunInfo is DockerRunInfo ? targetService.Name : defaultHost);
+
+                // Review: should we split this codepath based on the KIND of service that 
+                // provides the binding?
```

Comment:
@davidfowl  Let's discuss this.

I don't really have  strong feeling. What I wonder is if we should **avoid** populating the HOST/PORT env-vars in dev because they won't be there in production. However, the example that I liked about with connection string uses token replacement based on the env-vars, so..... 🤣

Reply:
I think it's worth a shot, though if we do container and not container, it'll break in the case where you run with docker.